actors/miner/src/lib.rs

```diff
@@ -2426,15 +2430,7 @@ impl Actor {
 
                     deadline_power_delta += &partition_power_delta;
                     deadline_pledge_delta += &partition_pledge_delta; // expected to be zero, see note below.
-
-                    // daily_fee should not change when replacing sectors with updated versions of themselves
-                    if partition_daily_fee_delta != TokenAmount::zero() {
-                        return Err(actor_error!(
-                            illegal_state,
-                            "unexpected daily fee change when extenting sectors at {:?}",
-                            key
-                        ));
-                    }
+                    deadline_daily_fee_delta += &partition_daily_fee_delta; // non-zero when extending legacy sectors
```

Comment:
nit: by legacy here you mean pre-fee sectors, something different than `ExtensionKind::ExtendCommittmentLegacy`.  Clarifying this avoids some confusion